zebra-network/src/peer_set/initialize.rs

```diff
@@ -114,16 +116,26 @@ where
         inv_receiver,
     );
     let peer_set = Buffer::new(BoxService::new(peer_set), constants::PEERSET_BUFFER_SIZE);
+    info!("checkpoint: {}", line!());
 
-    // Connect the tx end to the 3 peer sources:
+    let initial_peers_fut = {
+        let initial_peers = config.initial_peers();
```

Comment:
DNS can take a long time, should we wrap initial_peers in spawn_blocking?
That way, it gets its own thread, and doesn't block other tasks.

Are we allowed to call spawn_blocking here?
(Check the documentation.)

Reply:
Let's also log an info-level message outside the spawn_blocking, to help us diagnose bugs.

Reply:
Since tokio uses cooperative multitasking, and the scheduler is non-deterministic, a blocking function can block a spawned task, even if that task is spawned first.

So let's fix the blocking DNS, then see if the listener test passes on macOS?

We should replace all the calls to `to_socket_addrs()` with `tokio::net::lookup_host().await?`.
https://docs.rs/tokio/1.1.0/tokio/net/fn.lookup_host.html

We should also apply a timeout of at least 5 seconds to each DNS lookup:
https://tools.ietf.org/rfcmarkup?doc=1123#section-6.1.3.3

Reply:
(I've opened ticket #1631 to make a similar change for the listener configs. We don't need to do that in this PR, because they're all IP addresses in the tests, so they can't hang.)